mailet/standard/src/main/java/org/apache/james/transport/mailets/StripAttachment.java

```diff
@@ -346,71 +362,97 @@ private boolean checkMessageRemoved(Part part, Mail mail)
+    }
+
     /**
-     * Saves the content of the part to a file in the given directoy, using the
+     * Saves the content of the part to a file in the given directory, using the
      * name of the part. If a file with that name already exists, it will
      * 
      * @param part
      *            The MIME part to save.
      * @return
      * @throws Exception
      */
-    private String saveAttachmentToFile(Part part, String fileName)
-            throws Exception {
-        BufferedOutputStream os = null;
-        InputStream is = null;
-        File f = null;
+    @VisibleForTesting String saveAttachmentToFile(Part part, Optional<String> fileName) throws Exception {
         try {
-            if (fileName == null)
-                fileName = part.getFileName();
-            int pos = -1;
-            if (fileName != null) {
-                pos = fileName.lastIndexOf(".");
-            }
-            String prefix = pos > 0 ? (fileName.substring(0, pos)) : fileName;
-            String suffix = pos > 0 ? (fileName.substring(pos)) : ".bin";
-            while (prefix.length() < 3)
-                prefix += "_";
-            if (suffix.length() == 0)
-                suffix = ".bin";
-            f = File.createTempFile(prefix, suffix, new File(directoryName));
-            log("saving content of " + f.getName() + "...");
-            os = new BufferedOutputStream(new FileOutputStream(f));
-            is = part.getInputStream();
-            if (!(is instanceof BufferedInputStream)) {
-                is = new BufferedInputStream(is);
-            }
-            int c;
-            while ((c = is.read()) != -1) {
-                os.write(c);
-            }
+            File outputFile = outputFile(part, fileName);
+
+            log("saving content of " + outputFile.getName() + "...");
+            IOUtils.copy(part.getInputStream(), new FileOutputStream(outputFile));
 
-            return f.getName();
+            return outputFile.getName();
         } catch (Exception e) {
-            log("Error while saving contents of ["
-                    + (f != null ? f.getName() : (part != null ? part
-                            .getFileName() : "NULL")) + "].", e);
+            log("Error while saving contents of", e);
             throw e;
-        } finally {
-            is.close();
-            os.close();
         }
     }
 
+    private File outputFile(Part part, Optional<String> fileName) throws MessagingException, IOException {
+        if (!fileName.isPresent()) {
+            return createTempFile(part.getFileName());
+        }
+        return createTempFile(fileName.get());
+    }
+
+    private File createTempFile(String originalFileName) throws IOException {
+        OutputFileName outputFileName = OutputFileName.from(originalFileName);
+        return File.createTempFile(outputFileName.getPrefix(), outputFileName.getSuffix(), new File(directoryName));
+    }
+
+    private static class OutputFileName {
```

Comment:
I don't understand the logic in this class with the crazy going on with **prependedPrefix**. Please make it public, and test it.